src/stream.c

```diff
@@ -340,6 +340,9 @@ static avifResult makeRoom(avifRWStream * stream, size_t size)
     size_t neededSize = stream->offset + size;
     size_t newSize = stream->raw->size;
     while (newSize < neededSize) {
+        if (newSize > SIZE_MAX - AVIF_STREAM_BUFFER_INCREMENT) {
+            return AVIF_RESULT_OUT_OF_MEMORY;
+        }
```

Comment:
If `size_t` is 64 bits, the test for this new overflow check takes a very long time because `AVIF_STREAM_BUFFER_INCREMENT` is `(1024 * 1024)`.

Reply:
We could replace this loop with a formula:
- It would be slower for the general case but that should be negligible compared to calling the codec anyway,
- It would speed the test up.

Formula could be something like

```c
    AVIF_CHECKERR(size <= SIZE_MAX - stream->offset, AVIF_RESULT_OUT_OF_MEMORY);
    if (stream->raw->size >= stream->offset + size) return AVIF_RESULT_OK; // big enough already
    size_t neededExtra = size == AVIF_STREAM_BUFFER_INCREMENT ? 0 : AVIF_STREAM_BUFFER_INCREMENT - size % AVIF_STREAM_BUFFER_INCREMENT;
    AVIF_CHECKERR(stream->offset + size <= SIZE_MAX - neededExtra, AVIF_RESULT_OUT_OF_MEMORY);
    return avifRWDataRealloc(stream->raw, stream->offset + size + neededExtra);
```